### ISSUE TITLE
Error when pushing : "error: RPC failed; HTTP 401 curl 22 The requested URL returned error: 401"

### ISSUE THREAD

**Issue by nicos68** [gogits/gogs#4718]
- Gogs version (or commit ref): 0.11.29.0727
- Git version: 2.11.0
- Operating system: Kubuntu 17.04 (client), Raspberry Pi docker image (server)
- Database:
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist (usually found in `log/gogs.log`):
[logs.txt](https://github.com/gogits/gogs/files/1247350/logs.txt)
## Description
I am running a gogs server on my raspberry pi proxy using the gogs-rpi docker image. An NGINX reverse proxy is set in front of the server.
I am able to create a repo and clone it, then I create a file and commit it. When I run `git push` I get the folloing error:
`error: RPC failed; HTTP 401 curl 22 The requested URL returned error: 401`
`fatal: The remote end hung up unexpectedly`
`fatal: The remote end hung up unexpectedly`

**Comment by nicos68** [comment]
wrong pass word ... :blush:

**Comment by phamthanhnhan14** [comment]
Git clone working well when git push get 401 error

**Comment by abdennour** [comment]
Any news about this bug ?

**Comment by icodemo** [comment]
i had simillar problem with my gogs instance

**Comment by alexisCata** [comment]
Still same issue here.
Tried:
Password checked and correct
git config --global --unset http.postBuffer
git config --global http.postBuffer 1048576000
Checked window credential manager

**Comment by JayHoltslander** [comment]
Same issue. Password is correct.

**Comment by raandree** [comment]
I have the same issue with TFS and Git 2.21.0. It seems to be an issue with Git and the size of the the repo / of some change. The password is correct for sure. At least it works with a small test repo so it cannot be the password.

**Comment by davidalbertonogueira** [comment]
I am able to overcome this error on every new repo by setting the origin url to include the username:
`http://<mygitserver>:<gitserverport>/<RepoName>.git
->
http://<myuser>@<mygitserver>:<gitserverport>/<RepoName>.git`

**Comment by silverua** [comment]
Had the same issue, password was 100% correct. The symptoms that I had were - only one of my computers was affected. I could pull and clone without restrictions, but I could only push super small chunks like code only. Big commits that contained graphics or sound files always returned this error.
Tried to reinstall Git and Git clients on the affected PC. That did not help.
What helped was:
git config --global --unset http.postBuffer

**Comment by mhshd** [comment]
I had the same issue. I used these commands and it worked:
git remote remove origin
git remote add origin https://github..com/user/repo
git push --set-upstream origin master

source : [https://stackoverflow.com/questions/15240815/git-fatal-the-remote-end-hung-up-unexpectedly](url)

**Comment by CoolSuplex** [comment]
@mhshd above commands did not worked for me :(

**Comment by uabutler** [comment]
I'll just leave this here for anyone running into this problem in the future. I just had this problem because I forgot to turn off MFA. Worked just fine afterward.

**Comment by didoviktorov** [comment]
I had the same error, which occured after i've changed my username. None of the above suggestions solved my problem. After half a day of digging, i found this solution -> 

https://www.programmersought.com/article/36666013442/#:~:text=Privacy%20Policy-,%5BSolved%5D%20RPC%20failed%20HTTP%20401%20curl%2022%20The,requested%20URL%20returned%20error%3A%20401&text=Change%20the%20user%20name%20and,they%20are%20correct%2C%20it's%20ok!

The error was caused by the inconsistency between the local username and password and the online.
In short Enter the control panel of the computer => User Account => Manage Windows Credentials and find the website where the code repository is located.
Change the user name and password to be the same as those on the online code warehouse website. Submit the code again and you will be prompted to re-enter the user name and password. If they are correct, it’s ok!

**Comment by LiQiongchao** [comment]
I had the same issue. I used ssh replace http and it worked.
